proxy/src/http/server.rs

```diff
@@ -8,9 +10,30 @@ async fn status_handler(_: Request<Body>) -> Result<Response<Body>, ApiError> {
     json_response(StatusCode::OK, "")
 }
 
+/// Process a session kick callback from the control plane. The body is a
+/// KickSession as a JSON document.
+///
+/// TODO: authentication
+async fn kick_session_handler(req: Request<Body>) -> Result<Response<Body>, ApiError> {
+    let body = &body::to_bytes(req.into_body())
+        .await
+        .context("Failed to get request body")
+        .map_err(ApiError::BadRequest)?;
+    let kick_session_json: KickSession = serde_json::from_slice(body)
+        .context("Failed to parse query as json")
+        .map_err(ApiError::BadRequest)?;
+
+    match auth::backend::notify(kick_session_json.session_id, Ok(kick_session_json.result)) {
+        Ok(()) => json_response(StatusCode::OK, ""),
+        Err(NotifyError::NotFound(s)) => Err(ApiError::NotFound(anyhow::anyhow!(s))),
+        Err(e @ NotifyError::Hangup) => Err(ApiError::NotFound(anyhow::anyhow!(e))),
+    }
+}
```

Comment:
```rust
auth::backend::notify(kick_session_json.session_id, Ok(kick_session_json.result))
    .map_err(|e| ApiError::NotFound(anyhow::anyhow!(e)))?;

json_response(StatusCode::OK, "")
```

Reply:
There's two more action points here:

### The payload

The console should be able to pass an error, not just a "happy path" json. Otherwise it's impossible to notify the user that something bad happened, because there's no message for that.

For instance, we could use this model:

```rust
#[derive(Deserialize)]
#[serde(tag = "kind")] // 
enum Kick {
  Success(DatabaseInfo),
  Failure {
    code: u32, // the code will allow us to compose a message to be shown to user.
    description: String, // some strings might be sensitive and not suitable for forwarding.
  },
}
```

It should be possible to represent this type in an openapi spec, also see this https://serde.rs/enum-representations.html.

cc @ololobus

### OpenAPI spec

Once we implement something non-trivial, it'd be great to have an OpenAPI spec. I've been thinking of using https://github.com/juhaku/utoipa, but you definitely don't have to do this right now in this PR.